src/android/FileTransfer.java

```diff
@@ -528,14 +528,18 @@ public void run() {
                         }
                         safeClose(inStream);
                     }
-                    
+
                     Log.d(LOG_TAG, "got response from server");
                     Log.d(LOG_TAG, responseString.substring(0, Math.min(256, responseString.length())));
-                    
+
                     // send request and retrieve response
                     result.setResponseCode(responseCode);
                     result.setResponse(responseString);
 
+                    if(result.getBytesSent() != totalBytes) {
```

Comment:
This code is related to [the added test](https://github.com/apache/cordova-plugin-file-transfer/pull/117/files#diff-2a8a5fef3397df87ab538f028a5c6b50R951).